Module 1.2/README.md

```diff
@@ -0,0 +1,411 @@
+
+	- Parsing: Una vez ya tokenizado todo el código se genera un árbol con todos los tokens generados que pertenecerán al lenguaje, este árbol generado se llama Abstract Syntax Tree , y es de donde se genera el código más básico entendido por la computadora. 
+
+	- Code Generation: Se ejecuta todo el código generado por el AST, para que la computadora vaya realizando accione y reservando memoria. 
+
+JS siendo un lenguaje compilado, existen 3 formas de validar esto, que es con la generación de errores que salen antes de ejecutar el condigo, dando así cabida a la idea de “compila luego ejecuta”. Los tipos de errores que salen para verificar dicha afirmación son: 
+
+	- syntax errors 
+
+	- early errors 
+
+	- hoisting 
+
+* Compiler Speak 
+
+Al momento de compilar el código, existen múltiples operaciones que realiza el compilador en sí, entre las más importantes están las operaciones de derecha y las operaciones de izquierda. Estas operaciones hacen referencia a los dos lados que existen al momento de usar “=”, también son conocidas con sus nombres en inglés como “LHS” y “RHS”. Estos pueden ser tomados como fuentes (que proporcionan un valor) o como objetivos (que sirven para almacenar los datos que proporcionan las fuentes), cabe recalcar que no toda LHS puede ser un target y no toda RHS puede ser un source. 
+
+	- Target(objetivo): se puede tomar a esto como una asignación a una variable, para que 	esta almacene algún tipo de valor en memoria, no toda asignación de variable esta tan clara 	como al momento de usar “=” algunas se encuentran de manera más implícita, como al 		momento de usar una iteración sobre un objeto. 
+
+	- Source(fuente): viene a ser la raíz de donde saldrá todo valor para un target, la manera 	más sencilla de entender esto es, que todo aquelle que devuelva algún tipo de valor que pueda 	ser referenciado será un “source”. 
+
+* Cheating: Runtime Scope Modifications 
+
+Tenemos que tener en claro que el scope es generado al momento de la compilación, eso quiere decir que no se tiene un acceso directo al scope, para hacer cualquier tipo de manipulación sobre este. En JS existen 2 formas de acceder a el scope una es hacer uso de la función “eval()” y otra hacer uso de la palabra reservada “with”. 
+
+La función “eval()” lo que hace es recibir un parámetro para ser evaluado, lo que quiere decir que será compilado al mismo tiempo de la compilación del código, lo que quiere decir que, si existe algún tipo de declaración dentro de esta función, se generar un cambio en el scope afectando a el tiempo d ejecución. 
+
+Por otro lado, la palabra reservada “with” lo que hace es afectar directamente al scope, cambiándolo por un nuevo objeto, lo cual afecta grande mente al tiempo de ejecución y al desempeño de lo que se esté haciendo. 
+
+* Lexical Scope 
+
+Las variables pertenecen a un scope, donde estas existen y se pueden acceder, dichos scopes son asignados al momento de la compilación lo que se conoce como “lexical scope”. 
```

Comment:
I will steal your notes 💯 , I liked that you put the important points in big letters and the examples reflect exactly the concept of the topics.

Reply:
Thank you so much.